packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -212,20 +225,19 @@ export class SearchBar extends React.PureComponent {
             onBlur={this.handleBlur}
             onKeyDown={this.handleKeyPress}
             className={inputClassName}
+            data-element="search-input"
+          />
+          <CloseIcon
+            width="18px"
+            height="18px"
+            onClick={shouldDisplayClearButton ? this.handleClear : null}
+            onKeyDown={
+              shouldDisplayClearButton ? this.handleClearButtonKeyDown : null
+            }
+            className={closeIconClassName}
+            tabIndex="0"
+            data-element="close-icon"
```

Comment:
I was thinking about that recently and read a couple of articles about e2e tests and selectors. 
`className` is guaranteed to not change in design-system lib but maybe we should think about this property just as a way to style things. `data-testId` could inform about what is this element and it could be used in tests. Also there is a smaller chance that it will change (than class name). It will be the same until the element is removed or replaced with something else.

To sum it up - let's add `data-` attribute but it would be better to use the convention proposed in AA + use `button` word instead of `icon` since it's clickable:
```suggestion
            data-testId="close-button"
```

Reply:
I'm all in for that approach :)